src/commands.js

```diff
@@ -358,17 +358,77 @@ function toggleHeader(type) {
   }
 }
 
+function isHeaderEnabledByType(type, rect, types) {
+  // Get cell positions for first row or first column
+  const cellPositions = rect.map.cellsInRect({
+    left: 0,
+    top: 0,
+    right: type == "row" ? rect.map.width : 1,
+    bottom: type == "column" ? rect.map.height : 1,
+  })
+  const map = rect.map
+
+  for (let i = 0; i < cellPositions.length; i++) {
+    const cell = rect.table.nodeAt(cellPositions[i])
+    if (cell && cell.type !== types.header_cell) {
+      return false
+    }
+  }
+
+  return true
+}
+
+// :: (string, ?{ useDeprecatedLogic: bool }) → (EditorState, dispatch: ?(tr: Transaction)) → bool
+// Toggles row|column header cells in a table always on the first row|column
+// to use the old, not support anymore, logic pass useDeprecatedLogic as true
+export function toggleHeader(type, options) {
+  options = options || { useDeprecatedLogic: false }
+
+  if (options.useDeprecatedLogic)
+    return deprecated_toggleHeader(type)
+
+  return function(state, dispatch) {
+    if (!isInTable(state)) return false
+    if (dispatch) {
+      let types = tableNodeTypes(state.schema)
+      let rect = selectedRect(state), tr = state.tr
+      let isHeaderEnabled = type === "column" ? isHeaderEnabledByType("row", rect, types) :
```

Comment:
Why we are passing `row` when it's `column` and vice versa?

Reply:
Because will need to know if the column/row header is enabled to select the right cells, for example:

Given I have 3x3 table with row header enabled
When I click to toggle the column header
Then I should not toggle the first cell since the row header is enabled